internal/environment/detect.go

```
@@ -85,49 +85,102 @@ func GetVersion(pType types.ProxmoxType) (string, error) {
 	}
 
 	debugPath := writeDetectionDebug()
 	if debugPath != "" {
-		return types.ProxmoxUnknown, "unknown", fmt.Errorf("unable to detect Proxmox environment (debug saved to %s)", debugPath)
+		return info, fmt.Errorf("unable to detect Proxmox environment (debug saved to %s)", debugPath)
+	}
+	return info, fmt.Errorf("unable to detect Proxmox environment")
+}
+
+// detectProxmox is retained as a compatibility wrapper for legacy call sites and tests.
+func detectProxmox() (types.ProxmoxType, string, error) {
```

golangci-lint (2.11.4)

[error] func detectProxmox is unused

(unused)